### PR TITLE
Fix doxygen param formatting

### PR DESCRIPTION
**Pull request by owen-mc-diffblue**
Fix formatting errors in doxygen function comments. I've tried to keep things in the right commits, but some might have slipped through.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [n/a] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [n/a] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [n/a] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [n/a] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [n/a] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: jbmc/src/java_bytecode/java_bytecode_convert_class.cpp [hunk 1]
```diff
@@ -1036,7 +1036,7 @@ static void find_and_replace_parameter(
 /// `java::Outer::T` instead of `java::Outer$Inner::T`.
 /// \param type
 /// \param replacement_parameters
-static void find_and_replace_parameters(
+static: void find_and_replace_parameters(
```
Comment (tautschnig) [inline jbmc/src/java_bytecode/java_bytecode_convert_class.cpp]: It seems the scripting got something wrong here, which is caused by the missing documentation of the parameters.

File: jbmc/src/java_bytecode/java_bytecode_convert_class.cpp [hunk 2]
```diff
@@ -1120,7 +1120,7 @@ void convert_java_annotations(
 /// corresponding outer classes.
 /// \param class_name
 /// \param symbol_table
-void mark_java_implicitly_generic_class_type(
+void: mark_java_implicitly_generic_class_type(
```
Comment (tautschnig) [inline jbmc/src/java_bytecode/java_bytecode_convert_class.cpp]: As above: documentation missing

**Review by tautschnig — APPROVED** [review]
Thanks a lot for all this cleanup work! The below changes are strictly required, but CI will tell you anyway and thus approving right away.

**Comment by owen-mc-diffblue** [comment]
I applied the fixes for @tautschnig's comments (I should have been more careful with my regex), removed some unneeded empty lines between functions that clang-format was complaining about and some overly long lines that the linter was complaining about. It now passes all the quick checks, so it should be good to go.

**Comment by owen-mc-diffblue** [comment]
@tautschnig I added four more commits, because I noticed some more easily fixable formatting problems

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 8d2912dc2b5e60c2b9003c9d8fae3fd485f09607).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/95618858
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 7c364c1e93b577e7c6d65c9f3286e17ff9ebc232).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/95619117

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 1957e4d7c225553c67eebea9c91b61041a02a48e).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/95621342

File: jbmc/src/java_bytecode/remove_exceptions.cpp [hunk 3]
```diff
@@ -260,13 +260,13 @@ void remove_exceptionst::instrument_exception_handler(
...
 /// is reversed whereas the letter ordering remains the same.
-/// @param stack_catch exception table
-/// @param goto_program program being evaluated
-/// @param[out] universal_try returns the try block corresponding to the
+/// \param stack_catch exception table
+/// \param goto_program program being evaluated
+/// \param[out] universal_try returns the try block corresponding to the
 ///   desired exception handler
-/// @param[out] universal_catch returns the catch block corresponding to the
+/// \param[out] universal_catch returns the catch block corresponding to the
```
Comment (tautschnig) [inline jbmc/src/java_bytecode/remove_exceptions.cpp]: Should a space before `[out]` be added as well?

**Review by tautschnig — APPROVED** [review]

**Review by kroening — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 94180e8959723097ff9b53098391bb00af6fc762).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/95752931

**Comment by owen-mc-diffblue** [comment]
@cesaro @smowton @thk123 As code owners, can you review this PR, which should be uncontroversial? Then I will fix up the conflicts and merge when CI passes.

**Review by smowton — APPROVED** [review]
Only reviewed insofar as I've scrolled past it at high speed and it looks like all the changes are small and not in code!

**Comment by owen-mc-diffblue** [comment]
I can confirm the changes are only in function comments, and are fairly small.

**Review by cesaro — APPROVED** [review]
Looks good to me!

**Review by chrisr-diffblue — APPROVED** [review]
LGTM

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 0437520c842aa456ee4148799cf1e88e1f593546).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/96180350
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Comment by owen-mc-diffblue** [comment]
I thought I had enough review now, but apparently not. It's a bit opaque to me, but it seems I need @martin-cs, @peterschrammel and/or @romainbrenguier.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 375b680440652bc85a4cdab18324e5562828642e).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/96205116

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**